kaitaistruct.py

```diff
@@ -293,6 +385,21 @@ def read_bytes(self, n):
             )
         return r
 
+    def write_bytes(self, data, pad=0, padding=b'\0'):
```

Comment:
You should follow the established approach from the Java runtime instead of devising a new one ([`ByteBufferKaitaiStream.java:532-555` on branch `serialization`](https://github.com/kaitai-io/kaitai_struct_java_runtime/blob/720ab6e6ac7b8875f4c56fe9e068f07f32b31d89/src/main/java/io/kaitai/struct/ByteBufferKaitaiStream.java#L532-L555)):

```java
    /**
     * Writes given byte array to the stream.
     * @param buf byte array to write
     */
    @Override
    public void writeBytes(byte[] buf) {
        bb.put(buf);
    }

    @Override
    public void writeBytesLimit(byte[] buf, long size, byte term, byte padByte) {
        int len = buf.length;
        bb.put(buf);
        if (len < size) {
            bb.put(term);
            long padLen = size - len - 1;
            for (long i = 0; i < padLen; i++)
                bb.put(padByte);
        } else if (len > size) {
            throw new IllegalArgumentException(
                    "Writing " + size + " bytes, but " + len + " bytes were given"
            );
        }
    }
```

The reason is that this method is already used by the compiler, and not only from the language specific part ([`JavaCompiler:834-835` on `serialization`](https://github.com/kaitai-io/kaitai_struct_compiler/blob/cfba0cccbdfa9b7ca77e498f9895ee6ecd2d7bfb/shared/src/main/scala/io/kaitai/struct/languages/JavaCompiler.scala#L834-L835)):

```scala
  override def attrBytesLimitWrite(io: String, expr: String, size: String, term: Int, padRight: Int): Unit =
    out.puts(s"$io.writeBytesLimit($expr, $size, (byte) $term, (byte) $padRight);")
```

but also from the common one ([`EveryWriteIsExpression:169` on `serialization`](https://github.com/kaitai-io/kaitai_struct_compiler/blob/cfba0cccbdfa9b7ca77e498f9895ee6ecd2d7bfb/shared/src/main/scala/io/kaitai/struct/languages/components/EveryWriteIsExpression.scala#L169)):

```scala
    attrBytesLimitWrite(io, expr, translator.translate(blt.size), term, padRight)
```

So when the Python serialization support in the compiler will be introduced, this `write_bytes` method that you wrote will have to be rewritten anyway.

Moreover, please follow the convention used in the rest of the Python library, that is: always accept a single byte as an integer number (`int`), instead of the `bytes` type in python. We've already discussed this at https://github.com/kaitai-io/kaitai_struct_python_runtime/pull/48#discussion_r465999296.

Ideally, just take the existing methods `writeBytes` and `writeBytesLimit` from Java runtime (shown above) and convert them to Python, with the exact same parameters and logic (if possible). You save yourself a lot of unnecessary work of inventing the new approach (and then eliminating bugs of this new logic), and we get a consistent system of methods across our runtime libraries.